app/controllers/appointments_controller.rb

```diff
@@ -131,4 +131,12 @@ def redirect_after_successful_update
       redirect_to calendar_path, success: 'Appointment has been modified'
     end
   end
+
+  def copy_appointment
+    return nil unless params[:copy_from].present?
+    appointment = Appointment.find(params[:copy_from]).dup
+    appointment.start_at = nil
+    appointment.end_at = nil
+    appointment
```

Comment:
The other stuff isn't rendered as inputs in the view, so they can't get saved. The only two afaik are those, cause they're actually hidden fields in the view.